patches/server/0840-Block-Ticking-API.patch

```diff
@@ -0,0 +1,50 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Owen1212055 <23108066+Owen1212055@users.noreply.github.com>
+Date: Sun, 26 Dec 2021 13:23:46 -0500
+Subject: [PATCH] Block Ticking API
+
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/block/CraftBlock.java b/src/main/java/org/bukkit/craftbukkit/block/CraftBlock.java
+index 7cf7b845620ab5b118da79f6057c0d80a254585e..077bd8670f036a32a5a9bafebcb77c120714e6f4 100644
+--- a/src/main/java/org/bukkit/craftbukkit/block/CraftBlock.java
++++ b/src/main/java/org/bukkit/craftbukkit/block/CraftBlock.java
+@@ -521,6 +521,23 @@ public class CraftBlock implements Block {
+         // SPIGOT-6778: Directly call setBlock instead of setTypeAndData, so that the tile entiy is not removed and custom remove logic is run.
+         return this.world.setBlock(position, fluidState.createLegacyBlock(), 3) && result; // Paper - leave liquid if waterlogged
+     }
++    // Paper start - Block tick API
++    @Override
++    public void tick() {
++        net.minecraft.world.level.block.state.BlockState blockData = this.getNMS();
++        var level = this.world.getMinecraftWorld();
++
++        blockData.getBlock().tick(blockData, level, position, level.random);
++    }
++
++    @Override
++    public void randomTick() {
++        net.minecraft.world.level.block.state.BlockState blockData = this.getNMS();
++        var level = this.world.getMinecraftWorld();
++
++        blockData.getBlock().randomTick(blockData, level, position, level.random);
++    }
++    // Paper start - Block tick API
```

Comment:
nit: That's probably supposed to be Paper end instead

Reply:
I thought I removed that! Whoops!